### DISCUSSION TITLE
Has anyone used Markdoc for multiple white-labeled documentation instances from a single single of content?

### DISCUSSION THREAD

**Discussion by lcriscuolo** [markdoc/markdoc#315] [General]
I am new to Markdoc, though I've watched the Stripe videos on their path to using it. I am exploring new documentation tooling options.

I have a mandatory use case in which documentation tooling support white-labeled our technical documentation for both product OEMs and partners. For this to be scalable and feasible for maintenance, we need to single-source the content as possible (similar to XML methodologies like DITA and DocBook), then use conditions, variables, functions and other means to customize that single source to display separately branded content to each OEM/partner.

Has anyone used Markdoc in this capacity? I realize that this was not one of Stripe's use cases when they developed Markdoc since they were developing it for their own use, and had no need to render multiple iterations of their content.

Any help would be appreciated. Thank you.

**Comment by nkohari-stripe** [comment]
Hey there! We actually have a team which is experimenting with customizing content in this way for platform customers; that is, using terminology that the platforms' users would understand, rather than Stripe's own terminology. Our main documentation content is also customized quite a bit for users who are browsing while logged in. While you're right that it isn't our primary use case, I think Markdoc would be a great fit.

You would still need to manage the visual theming separately in the React/HTML output that you'd be generating with Markdoc, but conditions and variables should be sufficient for customization of content specifically.

**Comment by lcriscuolo** [comment]
Thank you for your reply! From a high-level conceptual perspective, I wonder if it would be plausible to maintain separate Markdoc instances, and in each there is no "written content," but only use a document skeleton that references the single-source of content using partials? That way, in each instance, the writer can use variables and conditions to specify which (custom) content displays in each Markdoc instance?

I don't have enough understanding of Markdoc to understand how that works. I have used AsciiDoc in the past. Using AsciiDoc, one would use a pre-processor like AsciiDoctor to process and then render/build a document. But since Stripe uses their content from one instance that renders in real-time, I don't know how that would work in this scenario.

**Reply by mfix-stripe** [reply]
@lcriscuolo Markdoc is similar to AsciiDoc (here are some docs on why Markdoc instead of AsciiDoc: https://markdoc.dev/docs/faq#why-not-asciidoc)

Markdoc supports [variables](https://markdoc.dev/docs/variables), [functions](https://markdoc.dev/docs/functions), [conditionals](https://markdoc.dev/docs/tags#if/else), out of the box, and allows for other methods of extensibility (custom `transform` methods for [`nodes`](https://markdoc.dev/docs/nodes) and [`tags `](https://markdoc.dev/docs/tags).

As a last resort, you can even create your own [renderer](https://markdoc.dev/docs/render) (which in reality is just a function that takes the JSON-represented renderable-tree and outputs whatever you want).

In short, Markdoc is highly flexible, so I am sure there are ways to separate your content from the implementation in the ways you are describing. Keeping content separate from engineering implementation was a strict goal in creating Markdoc in the first place.
